pkg/controller/scheduler.go

```diff
@@ -866,6 +873,12 @@ func (c *Controller) checkCanaryStatus(canary *flaggerv1.Canary, canaryControlle
 	}
 
 	var err error
+	// update canary status to ensure checksum is up to date in all webhooks
+	if err := canaryController.SyncStatus(canary, flaggerv1.CanaryStatus{Phase: flaggerv1.CanaryPhaseProgressing}); err != nil {
```

Comment:
I think this will break the whole thing, we can't set `CanaryPhaseProgressing` here, the canary status is synced only in `shouldAdvance`.

Reply:
@stefanprodan local integration tests are passing and also validated that the image generated by this change works on a dev cluster but  to your point I've moved the code block under `shouldAdvance`